addons/web/static/src/core/datetime/datetime_picker.js

```diff
@@ -421,8 +421,8 @@ export class DateTimePicker extends Component {
             throw new Error(`DateTimePicker error: given "maxDate" comes before "minDate".`);
         }
 
-        const timeValues = this.values.map((val) => [
-            (val || DateTime.local()).hour,
+        const timeValues = this.values.map((val,index) => [
```

Comment:
lint/prettify your code (you can enable the tooling via /addons/web/tooling/enable.sh)